functions/src/pdfBuilder/donationCertificateBuilderFunction.ts

```diff
@@ -0,0 +1,140 @@
+import * as functions from 'firebase-functions';
+//import { createAnduploadDonationCertificate } from './buildDonationCertificateCH';
+import { initializeApp } from "firebase/app";
+import { getFirestore, connectFirestoreEmulator } from 'firebase/firestore';
+import { doc, setDoc, getDocs, collection } from "firebase/firestore";
+import { createAndUploadDonationCertificate } from './generateDonationCertificatePDF';
+import * as config from '../config';
+
+let FIRESTORE_EMULATOR_REQUIRED = true;
+
+export const bulkDonationCertificateBuilderFunction = functions.https.onCall(async (data, context) => {
+	const app = await initializeFirebaseApp();
```

Comment:
I think you should be able to reuse the `firestore` singleton in the `useFirestoreAdmin.ts`:
https://github.com/socialincome-san/public/blob/614605df94c297aa62b56d6d53fe9a036e576067/functions/src/useFirestoreAdmin.ts#L16

Then you can remove the `initializeFirestore` and `initializeFirebaseApp` here.

Reply:
Could use the Singleton for Firestore, for the Storage it was not possible for me. Perhaps we can solve that in a second step :-)